src/test/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/createbundle/CreateBundleAboutToSubmitHandlerTest.java

```diff
@@ -287,12 +279,66 @@ public void givenCaseWithEditedDwpDocsAndPhmeUnderReview_thenReturnErrorMessageA
+        callback.getCaseDetails().getCaseData().setLanguagePreferenceWelsh("No");
+        if (party.equals("appellant")) {
+            callback.getCaseDetails().getCaseData().setConfidentialityRequestOutcomeAppellant(getDatedRequestOutcome(RequestOutcome.IN_PROGRESS));
+        } else {
+            callback.getCaseDetails().getCaseData().setConfidentialityRequestOutcomeJointParty(getDatedRequestOutcome(RequestOutcome.IN_PROGRESS));
+        }
+
+        addMandatoryNonEditedDwpDocuments();
+
+        PreSubmitCallbackResponse<SscsCaseData> response = handler.handle(ABOUT_TO_SUBMIT, callback, USER_AUTHORISATION);
+
+        assertEquals("There is a pending enhanced confidentiality request on this case", response.getErrors().iterator().next());
+        verifyNoInteractions(serviceRequestExecutor);
+    }
+
+    private DatedRequestOutcome getDatedRequestOutcome(RequestOutcome outcome) {
+        return DatedRequestOutcome.builder().requestOutcome(outcome).build();
+    }
+
+    @Test
+    @Parameters({"No, bundleEnglishEditedConfig, bundleEnglishConfig", "Yes, bundleWelshEditedConfig, bundleWelshConfig"})
+    public void givenEnhancedConfidentialityCaseWithEditedDocuments_thenPopulateEditedAndUneditedConfigFileName(String langPreference, String expectedBundleConfig1, String expectedBundleConfig2) {
+        addMandatoryNonEditedDwpDocuments();
+        addEditedSscsDocuments();
+
+        callback.getCaseDetails().getCaseData().setIsConfidentialCase(YesNo.YES);
+        callback.getCaseDetails().getCaseData().setLanguagePreferenceWelsh(langPreference);
+        PreSubmitCallbackResponse<SscsCaseData> response = handler.handle(ABOUT_TO_SUBMIT, callback, USER_AUTHORISATION);
+
+        verify(serviceRequestExecutor).post(callback, "bundleUrl.com/api/new-bundle");
+        assertEquals(2, response.getData().getMultiBundleConfiguration().size());
+        assertEquals(expectedBundleConfig1, response.getData().getMultiBundleConfiguration().get(0).getValue());
+        assertEquals(expectedBundleConfig2, response.getData().getMultiBundleConfiguration().get(1).getValue());
+    }
+
+    @Test
+    @Parameters({"Yes, bundleWelshConfig", "No, bundleEnglishConfig"})
+    public void givenEnhancedConfidentialityCaseWithNoEditedDocs_thenPopulateUneditedConfigFileName(String langPreference, String expectedBundleName) {
+        addMandatoryNonEditedDwpDocuments();
+
+        callback.getCaseDetails().getCaseData().setIsConfidentialCase(YesNo.YES);
+        callback.getCaseDetails().getCaseData().setLanguagePreferenceWelsh(langPreference);
+        PreSubmitCallbackResponse<SscsCaseData> response = handler.handle(ABOUT_TO_SUBMIT, callback, USER_AUTHORISATION);
+
+        verify(serviceRequestExecutor).post(callback, "bundleUrl.com/api/new-bundle");
+        assertEquals(1, response.getData().getMultiBundleConfiguration().size());
+        assertEquals(expectedBundleName, response.getData().getMultiBundleConfiguration().get(0).getValue());
+    }
+
```

Comment:
Do you think it's worth adding a unit test (or an integration test) that checks the scenario for when the case is confidential AND a PHME case?

Reply:
yes, I missed that.